src/aioswitcher/bridge.py

```diff
@@ -372,6 +391,39 @@ def get_shutter_direction(self) -> ShutterDirection:
         directions = dict(map(lambda d: (d.value, d), ShutterDirection))
         return directions[hex_direction]
 
+    # Switcher S12 methods
+
+    def get_s12_device_state(self) -> DeviceState:
+        """Extract the device state from the broadcast message."""
+        hex_device_state = hexlify(self.message)[270:272].decode()
+        return (
+            DeviceState.ON
+            if hex_device_state == DeviceState.ON.value
+            else DeviceState.OFF
+        )
+
+    def get_s12_position_one(self) -> int:
+        """Return the current position of the shutter 0 <= pos <= 100."""
+        hex_pos = hexlify(self.message[167:169]).decode()
+        return int(hex_pos[2:4]) + int(hex_pos[0:2], 16)
+
+    def get_s12_direction_one(self) -> ShutterDirection:
+        """Return the current direction of the shutter (UP/DOWN/STOP)."""
+        hex_direction = hexlify(self.message[169:171]).decode()
+        directions = dict(map(lambda d: (d.value, d), ShutterDirection))
+        return directions[hex_direction]
+
+    def get_s12_position_two(self) -> int:
+        """Return the current position of the shutter 0 <= pos <= 100."""
+        hex_pos = hexlify(self.message[151:153]).decode()
+        return int(hex_pos[2:4]) + int(hex_pos[0:2], 16)
+
+    def get_s12_direction_two(self) -> ShutterDirection:
+        """Return the current direction of the shutter (UP/DOWN/STOP)."""
+        hex_direction = hexlify(self.message[153:155]).decode()
+        directions = dict(map(lambda d: (d.value, d), ShutterDirection))
+        return directions[hex_direction]
```

Comment:
This would be better to handle with a parameter and keep the standard design, e.g., add option for the number of the controlled shutter, else it would be complicated to add this into the integration since function names needs to be different if we control the first or the 2nd shutter.